### PR TITLE
Adding toleration and affinity

### PR DESCRIPTION
**Pull request by severity1**
##### Description

Adding the ability to configure toleration and affinity to the setup job.

### CODE REVIEW

**Comment by severity1** [comment]
this fixes issue #2424

**Review by sumo-drosiek — COMMENTED** [review]
@severity1 Thank you for the contribution

It would be awesome if you add missing configuration to chart's README.md as well

**Comment by severity1** [comment]
> @severity1 Thank you for the contribution
It would be awesome if you add missing configuration to chart's README.md as well

Hi @sumo-drosiek, README.md has been updated.

**Review by sumo-drosiek — APPROVED** [review]
Thank you, I will fix the non-related markdown table changes in separate PR

**Comment by sumo-drosiek** [comment]
I won't as this is required check :sweat_smile:

File: deploy/helm/sumologic/README.md [hunk 1]
```diff
@@ -389,7 +391,7 @@ The following table lists the configurable parameters of the Sumo Logic chart an
...
+| `otelevents.statefulset`                                                  | OpenTelemetry Collector StatefulSet customization options. See values.yaml for more details.                                                                                                                                                                                                                                                                         | See values.yaml                                                                                                   |  | `Nil` |  | `Nil` |
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/README.md:394]:
```suggestion
| `otelevents.statefulset`                                                  | OpenTelemetry Collector StatefulSet customization options. See values.yaml for more details.                                                                                                                                                                                                                                                                         | See [values.yaml]                                                                                                   |
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,9 +15,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Changed
 
 [#2422]: feat(metadata): upgrade otelcol to v0.54.0-sumo-0 [#2422]
+[#2428]: fix(setup): add tolerations and affinity to setup job [#2428]
```
Comment (sumo-drosiek) [inline CHANGELOG.md:18]:
```suggestion
[#2428]: feat(setup): add tolerations and affinity to setup job [#2428]
```

File: deploy/helm/sumologic/README.md [hunk 3]
```diff
@@ -389,7 +391,7 @@ The following table lists the configurable parameters of the Sumo Logic chart an
...
+| `otelevents.statefulset`                                                  | OpenTelemetry Collector StatefulSet customization options. See values.yaml for more details.                                                                                                                                                                                                                                                                         | See [values.yaml]                                                                                                   |
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/README.md:394]: There was two additional spaces after my suggestion

```suggestion
| `otelevents.statefulset`                                                  | OpenTelemetry Collector StatefulSet customization options. See values.yaml for more details.                                                                                                                                                                                                                                                                         | See [values.yaml]                                                                                                 |
```

**Comment by sumo-drosiek** [comment]
@severity1 Thanks for quick reaction, we just need to reformat table. If you have some automation tool it would be better to use, otherwise I hope the latest suggestion will resolve the issue

File: deploy/helm/sumologic/README.md (hunk 3 above)
Comment (severity1) [reply deploy/helm/sumologic/README.md]: Oh, I missed this suggestion but I think I made a similar change just now and I still get formatting error, I ran this through my markdown formatter.

**Comment by severity1** [comment]
@sumo-drosiek, thank you for your patience with me, I ran my auto-formatter against the README.md and it still gives an error.

**Comment by sumo-drosiek** [comment]
@severity1 

Seems like headers are the issue, as I run `./ci/markdown_table_formatter.sh --format`, it replaced spaces to `-` locally

**Comment by severity1** [comment]
I think I found the issue

**Comment by severity1** [comment]
my IDE was converting `--|--` to `- | -`

**Comment by sumo-drosiek** [comment]
> my IDE was converting `--|--` to `- | -`

Yes, this was the issue. Apparently `./ci/markdown_table_formatter.sh` doesn't like it

**Comment by severity1** [comment]
thank you @sumo-drosiek cheers!

**Comment by sumo-drosiek** [comment]
Change has been released: https://github.com/SumoLogic/sumologic-kubernetes-collection/releases/v2.12.0

**Pull request merged**